File: frontend/src/lib/api/family.ts
import { http, jsonBody } from "./http";

export interface FamilyResponse {
  id: string;
  name: string;
  timezone: string;
  created_at: string;
}

export interface FamilyUpdateRequest {
  name?: string;
  timezone?: string;
}

/**
 * Household default language for the assistants. `auto` (default) → the
 * detect_language graph node runs per-turn detection from user input.
 * `en`/`pl` → seed default; per-turn detection still overrides on clear
 * opposite-language input. Drives the voice agent's greeting too.
 */
export type VoiceLocale = "auto" | "en" | "pl";

export interface FamilyPreferencesResponse {
  family_id: string;
  sync_interval_sec: number;
  fanout_enabled: boolean;
  voice_wake_enabled: boolean;
  always_on: boolean;
  auto_create_shopping_list: boolean;
  voice_locale: VoiceLocale;
  updated_at: string;
}

export type FamilyPreferencesPatch = Partial<
  Omit<FamilyPreferencesResponse, "family_id" | "updated_at">
>;

export const familyApi = {
  get: () => http<FamilyResponse>("/api/family"),
  patch: (body: FamilyUpdateRequest) =>
    http<FamilyResponse>("/api/family", {
      method: "PATCH",
      body: jsonBody(body),
    }),
  getPreferences: () =>
    http<FamilyPreferencesResponse>("/api/family/preferences"),
  patchPreferences: (body: FamilyPreferencesPatch) =>
    http<FamilyPreferencesResponse>("/api/family/preferences", {
      method: "PATCH",
      body: jsonBody(body),
    }),
};

File: backend/src/schemas/family.py
"""Family + FamilyPreferences request/response models."""
from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


# `auto` = detect per turn from user input; `en` / `pl` = seed default
# language. Used by `detect_language` graph node and the voice greeting.
VoiceLocale = Literal["auto", "en", "pl"]


class FamilyResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    timezone: str
    created_at: datetime


class FamilyUpdate(BaseModel):
    name: str | None = None
    timezone: str | None = None


class FamilyPreferencesResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    family_id: UUID
    sync_interval_sec: int
    fanout_enabled: bool
    voice_wake_enabled: bool
    always_on: bool
    auto_create_shopping_list: bool
    voice_locale: VoiceLocale
    updated_at: datetime


class FamilyPreferencesPatch(BaseModel):
    sync_interval_sec: int | None = None
    fanout_enabled: bool | None = None
    voice_wake_enabled: bool | None = None
    always_on: bool | None = None
    auto_create_shopping_list: bool | None = None
    voice_locale: VoiceLocale | None = None
